gosling/api.py

```diff
@@ -41,6 +43,59 @@ def properties(self, **kwargs):
         return copy
 
 
+class _TransformsMixen:
+    def _add_transform(self, *transforms):
+        copy = self.copy()
+        if copy.dataTransform is Undefined:
+            copy.dataTransform = []
+        copy.dataTransform.extend(transforms)
+        return copy
+
+    def transform_filter(self, field, **kwargs):
+        return self._add_transform(
+            core.FilterTransform(type="filter", field=field, **kwargs)
+        )
+
+    def transform_filter_not(self, field, **kwargs):
+        kwargs["not"] = True
+        return self._add_transform(
+            core.FilterTransform(type="filter", field=field, **kwargs)
+        )
```

Comment:
`not` is a reserved word in Python, so you can't use it as a keyword argument:

```python
def foo(not): # SyntaxError: invalid syntax
  return True
```

This is a workaround to set the field 'not' to `True`. Maybe a better name would be `transform_filter_except`? The other option would be to add `not_` to the `transform_filter` def:

```python
    def transform_filter(self, field, not_=Undefined, **kwargs):
        kwargs["not"] = not_
        return self._add_transform(
            core.FilterTransform(type="filter", field=field, **kwargs)
        )
```

Reply:
I like the first approach more as the second approach requires inconsistent `_` character. We could think about another term than `not` (e.g., `inverse`, etc) to avoid the appending character, but `transform_filter_except` looks good enough for me.